tools/image-registry.md

```diff
@@ -2,152 +2,155 @@
-* [TLS/SSL](image-registry.md#tlsssl)
-* [Authentication](image-registry.md#authentication)
+### Using with Docker CLI
 
-### Create Image Registry Service
+#### Login with Docker CLI
 
-#### Local Storage Backend
+#### Automated way
 
 ```
-$ kontena registry create
+$ kontena cloud image-repository docker-login
 ```
 
-#### Amazon S3 Storage Backend
+#### Manual way
 
-Write Amazon S3 access keys to Kontena Vault:
+* Create a permanent Kontena Cloud OAuth2 token:
 
-```
-$ kontena vault write REGISTRY_STORAGE_S3_ACCESSKEY <access_key>
-$ kontena vault write REGISTRY_STORAGE_S3_SECRETKEY <secret_key>
-```
+  ```
+  $ kontena cloud token create <NAME>
+  ```
 
-Create registry service:
+* Login with Docker CLI using your Kontena Cloud username and permanent token from the previous step:
 
-```
-$ kontena registry create --s3-bucket=<bucket_name> --s3-region=<optional_aws_region> --s3-v4auth
-```
+  ```
+  $ docker login -u <USERNAME> images.kontena.io
+  ```
 
-#### Azure storage backend
+#### Build & Push an image with Docker CLI
 
-Write Azure account key to Kontena Vault:
+Build:
 
 ```
-$ kontena vault write REGISTRY_STORAGE_AZURE_ACCOUNTKEY <azure_account_key>
+$ docker build -t images.kontena.io/<ORG>/<REPO>:<TAG> .
 ```
 
-Create registry service:
+Push:
 
 ```
-$ kontena registry create --azure-account-name=<account_name> --azure-container-name=<container_name>
+$ docker push images.kontena.io/<ORG>/<REPO>:<TAG>
 ```
 
-### Accessing Image Registry
+#### Pull an image with Docker CLI
 
-Before you can push images to the registry, you should set up the Kontena VPN service. In addition, you must set `--insecure-registry=registry.<grid_name>.kontena.local` in your local Docker daemon configuration.
+```
+$ docker pull images.kontena.io/<ORG>/<REPO>:<TAG>
+```
 
-Building and pushing an image to the registry:
+### List image repositories
 
 ```
-$ docker build -t registry.<grid_name>.kontena.local/myimage:mytag .
-$ docker push registry.<grid_name>.kontena.local/myimage:mytag
+$ kontena cloud image-repository list --org <ORG>
 ```
 
-Deploying an image from the registry:
+### Create a repository
 
 ```
-$ kontena service create myservice registry.<grid_name>.kontena.local/myimage:mytag
-$ kontena service deploy myservice
+$ kontena cloud image-repository create --org <ORG> <REPOSITORY>
 ```
 
-### TLS/SSL
-
-Generate your own certificate:
+### Show a repository details
 
 ```
-$ openssl req -x509 -newkey rsa:2048 -keyout registry_key.pem -out registry_ca.pem -days 1080 -nodes -subj '/CN=registry.<grid_name>.kontena.local/O=My Company Name LTD./C=US'
+$ kontena cloud image-repository show <REPOSITORY>
 ```
 
-Write key and certificate to Kontena Vault:
+### Remove a repository
 
 ```
-$ kontena vault write REGISTRY_HTTP_TLS_KEY "$(cat registry_key.pem)"
-$ kontena vault write REGISTRY_HTTP_TLS_CERTIFICATE "$(cat registry_ca.pem)"
+$ kontena cloud image-repository remove <REPOSITORY>
```

Comment:
Should we mention something that the layers might or might not get deleted? I'm just wondering how many questions we'll get like:
"I removed my repo, but when pushed another not all the layers were pushed."

Reply:
Idk, might be confusing also to read it here?

Reply:
Yeah, might be too confusing